### ISSUE TITLE
App crashed after add relation to "Users" content type

### ISSUE THREAD

**Issue by mnlbox** [strapi/strapi#802]
**Node.js version**:
9.8.0
**npm version**:
5.6.0
**Strapi version**:
3.0.0-alpha.11.1
**Operating system**:
Linux Mint 18.1

**Do you want to request a *feature* or report a *bug*?**
Bug
**What is the current behavior?**
This is my Test model (`/api/test/models/Test.setting.json`):
```
{
  "connection": "default",
  "collectionName": "test",
  "info": {
    "name": "test",
    "description": ""
  },
  "options": {
    "timestamps": true
  },
  "attributes": {
    "name": {
      "type": "string",
      "multiple": false
    },
    "file": {
      "collection": "file",
      "via": "related",
      "plugin": "upload"
    }
  }
}
```

And this is my Users model (`plugins/users-permissions/models/User.settings.json`):
```
{
  "connection": "default",
  "info": {
    "name": "user",
    "description": ""
  },
  "attributes": {
    "username": {
      "type": "string",
      "minLength": 3,
      "unique": true,
      "configurable": false,
      "required": true
    },
    "email": {
      "type": "email",
      "minLength": 6,
      "configurable": false,
      "required": true
    },
    "provider": {
      "type": "string",
      "configurable": false
    },
    "password": {
      "type": "password",
      "minLength": 6,
      "configurable": false,
      "private": true
    },
    "resetPasswordToken": {
      "type": "string",
      "configurable": false,
      "private": true
    },
    "role": {
      "model": "role",
      "via": "users",
      "plugin": "users-permissions",
      "configurable": false
    },
    "tests": {
      "collection": "test",
      "via": "user",
      "plugin": " "
    }
  },
  "collectionName": "users-permissions_user"
}
```

**If the current behavior is a bug, please provide the steps to reproduce the problem**
## Step 1:
* Create a new project
* Add new content type called `test`
* Add two field to it:
  * name: Text field
  * file: File field
* Save

## Step 2:
* Goto users content type
* add new field with relation type (n-1) each user has some files and each file belong to one user
* Save

## Step 3:
now server restart with this message on console:
```
[2018-03-18T18:23:08.042Z] debug GET /content-type-builder/models/user?source=users-permissions (20 ms)
[2018-03-18T18:23:08.075Z] debug GET /content-type-builder/checkTableExists/default/users-permissions_user (4 ms)
[2018-03-18T18:23:11.501Z] debug GET /content-type-builder/models/user?source=users-permissions (6 ms)
[2018-03-18T18:23:11.545Z] debug GET 6f9be7693e25b1cbd3cbc815c3402a80.svg (18 ms)
[2018-03-18T18:23:11.547Z] debug GET d508607c9c3b2e347cc55197db9cf033.svg (10 ms)
[2018-03-18T18:23:11.548Z] debug GET 5dc7af4c72e415cfeb6e9114eee05cd5.svg (10 ms)
[2018-03-18T18:23:11.550Z] debug GET aabf753784ebc9494f5029f7be7ed5c1.svg (11 ms)
[2018-03-18T18:23:20.589Z] debug GET 639ff4090162d220beda67a9fe87a233.svg (2 ms)
[2018-03-18T18:23:20.590Z] debug GET 1d51d8767683a24635702f720cda4e26.svg (5 ms)
[2018-03-18T18:23:31.907Z] debug GET 3e3d916f6b8fc9ed209414446b40949a.svg (2 ms)
[2018-03-18T18:23:49.645Z] info The server is restarting

[2018-03-18T18:23:49.645Z] debug PUT /content-type-builder/models/user (27 ms)
[2018-03-18T18:23:50.874Z] error Something went wrong in the model `User` with the attribute `tests`
[2018-03-18T18:23:50.876Z] error Cannot read property 'models' of undefined
```
**What is the expected behavior?**
server restarts and content type works as expected

**Comment by mnlbox** [comment]
@soupette Any news about this?

**Comment by forsetius** [comment]
I also encountered this bug. It seems there is general problem with many-to-one relations. I have it now with "Category has many Articles" and "Menu has many Articles". On the other note, one-to-many relationships have same problem too. I'm using the Admin to add the relations, perhaps the problem is due to adding the relation only at one side?

**Comment by lauriejim** [comment]
@mnlbox and @forsetius you are using SQL database ?

**Comment by forsetius** [comment]
No, I'm using Mongo

**Comment by LuisLoureiro** [comment]
I think this issue is somehow related with #886.
I've also found a similar problem when adding a relationship.
I tried the _dockerized_ version and couldn't reproduce this problem.

**Node.js version:**
9.11.1
**npm version:**
5.6.0
**Strapi version:**
3.0.0-alpha.11.3
**Database:**
MongoDB
**Operating system:**
Linux Manjaro 17.1.8

**Comment by XcrossD** [comment]
@LuisLoureiro Thanks for linking the issues up. I didn't notice this one before. And yes I think it is exactly the same problem. Seems like there are quite a few users facing this problem, hopefully it can be fixed soon.

**Comment by Aurelsicoko** [comment]
In this issue, the problem is that you've created a Many-to-one relation but the attribute in the `one` side is missing. The `Test.settings.json` model should look like this:
```json
{
  "connection": "default",
  "collectionName": "",
  "info": {
    "name": "test",
    "description": ""
  },
  "options": {
    "timestamps": true
  },
  "attributes": {
    "name": {
      "multiple": false,
      "type": "string"
    },
    "file": {
      "collection": "file",
      "via": "related",
      "plugin": "upload"
    },
    "user": {
      "model": "user",
      "via": "tests",
      "plugin": "users-permissions"
    }
  }
}
```

and the `User.settings.json` like this:
```json
{
  "connection": "default",
  "info": {
    "name": "user",
    "description": ""
  },
  "attributes": {
    "username": {
      "type": "string",
      "minLength": 3,
      "unique": true,
      "configurable": false,
      "required": true
    },
    "email": {
      "type": "email",
      "minLength": 6,
      "configurable": false,
      "required": true
    },
    "provider": {
      "type": "string",
      "configurable": false
    },
    "password": {
      "type": "password",
      "minLength": 6,
      "configurable": false,
      "private": true
    },
    "resetPasswordToken": {
      "type": "string",
      "configurable": false,
      "private": true
    },
    "role": {
      "model": "role",
      "via": "users",
      "plugin": "users-permissions",
      "configurable": false
    },
    "tests": {
      "collection": "test",
      "via": "user",
      "plugin": ""
    }
  },
  "collectionName": "users-permissions_user"
}
```

However, there is a similarity in both issues is that you have the `multiple` property. I don't know why because we're not using it...

**Comment by XcrossD** [comment]
I checked again and this behavior is the same in #886. I think the relations on the `one` side should be created automatically, but even if I wanted to create each of them manually I cannot do it because one of them has to be saved first, and saving causes error...

**Comment by XcrossD** [comment]
@Aurelsicoko I found the error. I manually adjust the settings to be what you listed out and remove `plugin` property from both sides. The server can start again without error. This is currently what my settings look like.
![product-setting](https://user-images.githubusercontent.com/6669029/39235393-5ed362c2-48a8-11e8-8027-687947848408.PNG)
![productspecification-settings](https://user-images.githubusercontent.com/6669029/39235399-60e0f3c2-48a8-11e8-9590-f28c98d3f028.PNG)

**Comment by Aurelsicoko** [comment]
@XcrossD Did you add the `plugin` property manually or it comes from the Content-Type Builder?

**Comment by Aurelsicoko** [comment]
Okay, I found the issue! It should be fixed https://github.com/strapi/strapi/pull/1052

**Comment by LuisLoureiro** [comment]
@Aurelsicoko As you recommended I updated my **strapi** version to 3.0.0-alpha.12.0.1 and tried again.
I started a new project with the default database configurations and created my first two content types: **Clubs** and **Teams**.
Everything was ok until I decided to relate the two content types. 
I chose Clubs to relate with Teams and when I selected Teams, in the right-side box, the header got blank. (Exactly the same behavior @XcrossD described in the point 3 of his initial post in the #886)
Server stopped and I couldn't restart it. I had to remove the relationship from the _Clubs.settings.json_ file to have `strapi start` running as expected.
This is how my _Clubs.settings.json_ and my _Teams.settings.json_ file were after creating the relationship:
![screenshot from 2018-04-27 15-14-19](https://user-images.githubusercontent.com/1401605/39367102-c52c3414-4a2d-11e8-929a-d06757f5f4c4.png)

I'm using **Firefox 59.0.2 (64-bit)**.

I decided to switch to **Chromium 65 (64-bit)** and everything seems to work.
Sounds like a browser problem, in my case.
I don't see any console error nor warning.
This problem might be more #886 related.

Hope this helps.
